web3/_utils/method_formatters.py

```diff
@@ -440,6 +444,11 @@ def apply_list_to_array_formatter(formatter: Any) -> Callable[..., Any]:
     RPC.eth_getFilterLogs: filter_result_formatter,
     RPC.eth_getLogs: filter_result_formatter,
     RPC.eth_getProof: apply_formatter_if(is_not_null, proof_formatter),
+    RPC.eth_getRawTransactionByBlockHashAndIndex: HexBytes,
+    RPC.eth_getRawTransactionByBlockNumberAndIndex: apply_formatter_if(
+        is_not_null,
+        transaction_formatter,
+    ),
```

Comment:
I believe this should be the same as line 447 since they both return the same response format: 
```suggestion
    RPC.eth_getRawTransactionByBlockNumberAndIndex: HexBytes,
```